sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -451,6 +451,21 @@ trait CharVarcharTestSuite extends QueryTest with SQLTestUtils {
         Seq(Row("char(5)"), Row("varchar(3)")))
     }
   }
+
+  test("SPARK-33992: char/varchar resolution in correlated sub query ") {
```

Comment:
`branch-3.0` has the same issue? The branch does not have `PaddingAndLengthCheckForCharVarchar` though.

Reply:
nit: remove the unnecessary space in the end: `in correlated sub query ")` -> `in correlated sub query")`

Reply:
yes